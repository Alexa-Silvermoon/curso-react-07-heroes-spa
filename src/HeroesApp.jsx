import { AuthProvider } from "./auth"
import { AppRouter } from "./router/AppRouter"


export const HeroesApp = () => {
  return (

    // seguridad de login y logout gracias a AuthContext.jsx y AuthProvider.jsx
    <AuthProvider>
      <AppRouter/>
    </AuthProvider>

    // <>
    //     <AppRouter/>
    // </>
  )
}

// context y reducer https://www.udemy.com/course/react-cero-experto/learn/lecture/19965058#questions