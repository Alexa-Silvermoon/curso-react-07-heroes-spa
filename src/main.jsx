import React from 'react'
import ReactDOM from 'react-dom/client'
import { BrowserRouter, HashRouter } from 'react-router-dom'
import { HeroesApp } from './HeroesApp'
import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(


  <React.StrictMode>

    {/* <BrowserRouter> */}
    <HashRouter>

      <HeroesApp/>

    </HashRouter>
    {/* </BrowserRouter> */}

  </React.StrictMode>
  
  /* <React.StrictMode>
    <HeroesApp />
  </React.StrictMode> */
)
