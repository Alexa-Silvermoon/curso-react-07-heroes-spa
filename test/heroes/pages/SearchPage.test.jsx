import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from "../../../src/heroes"

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({

    ...jest.requireActual('react-router-dom'), // desestructurar y exparsir todo lo que viene en la libreria
    useNavigate: () => mockedUseNavigate // pero solo se sobrescribira el useNavigate

}));

describe('pruebas en SearchPage.jsx', () => {

    beforeEach( () => jest.clearAllMocks() );

    test('debe de mostrarse correctamente con valores por defecto', () => {

        const { container } = render(

            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>
        );

        screen.debug();

        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrarse a Batman y el input con el valor del queryString', () => {

        render(

            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        );

        screen.debug();

        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('batman');

        const img = screen.getByRole('img');
        expect( img.src ).toContain('/assets/heroes/dc-batman.jpg');

    });

    test('debe de ocultar el div del "Search a Hero" si hay un heroe mostrandose en pantalla', () => {

        render(

            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        );

        const alert = screen.getByLabelText('div-search-hero'); // lo identifica gracias al aria-label en SearchPage.jsx
        expect( alert.style.display ).toBe('none'); // se mostro el heroe? caso de si ocultese

    });

    test('debe de mostrar el error "No Hay Resultados para" si no se encuentra el heroe', () => {

        render(

            <MemoryRouter initialEntries={['/search?q=superman123']}>
                <SearchPage/>
            </MemoryRouter>
        );

        screen.debug();

        const alert2 = screen.getByLabelText('div-no-resultados-para'); // lo identifica gracias al aria-label en SearchPage.jsx
        expect( alert2.style.display ).toBe(''); // se mostro un error? caso de que si muestrese

    });

    test('debe de llamar el navigate a la pantalla nueva por medio del formulario', () => {

        const inputValue = 'flash';

        render(

            <MemoryRouter initialEntries={['/search?q=flash']}>
                <SearchPage/>
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        // fireEvent.input( input, { target: { value: 'flash' } } );
        fireEvent.change( input, { target: { name: 'SearchText' ,value: inputValue } } );

        const form = screen.getByRole('form');
        fireEvent.submit( form );

        // expect( mockedUseNavigate ).toHaveBeenCalled(); // funciona
        expect( mockedUseNavigate ).toHaveBeenCalledWith( `?q=${ inputValue }` );

    });
});

// pruebas en SearchPage https://www.udemy.com/course/react-cero-experto/learn/lecture/29491680#questions
// pruebas en queryParameters https://www.udemy.com/course/react-cero-experto/learn/lecture/29491712?start=15#questions
// requireActual https://www.udemy.com/course/react-cero-experto/learn/lecture/29491766#questions
