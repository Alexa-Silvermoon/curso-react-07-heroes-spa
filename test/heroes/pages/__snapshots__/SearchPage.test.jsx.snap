// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en SearchPage.jsx debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Busqueda
  </h1>
  <hr />
  <div
    class="row"
  >
     
    <div
      class="col-5"
    >
      <h4>
        Buscar
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Busca aqui a tu heroe"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        aria-label="div-search-hero"
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a Hero
      </div>
      <div
        aria-label="div-no-resultados-para"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No Hay Resultados para 
        <b />
      </div>
    </div>
  </div>
</div>
`;
